### PR TITLE
[20.05] explicitly cast oidc providers to a list

### PR DESCRIPTION
**Pull request by almahmoud**
Without the explicit casting, that line returns `TypeError: 'dict_keys' object does not support indexing` in py3.
`.keys()` returns an indexable list in py3 but a non-idexable `dict_keys` in py3.
https://stackoverflow.com/a/17322707

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/controllers/root.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def login(self, trans, redirect=None, is_logout_redirect=False, **kwd):
                 len(trans.app.config.oidc) == 1 and
                 len(trans.app.auth_manager.authenticators) == 0
                 and is_logout_redirect is False):
-            provider = trans.app.config.oidc.keys()[0]
+            provider = list(trans.app.config.oidc.keys())[0]
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/root.py]: Not a performance issue here, but you can skip the (potentially expensive) list creation with:
```suggestion
            provider = next(iter(trans.app.config.oidc.keys()))
```

**Comment by nsoranzo** [comment]
Did you mean to target the 20.05 branch? It's targeting `dev` at the moment.

**Comment by almahmoud** [comment]
I did mean to target 20.05, that was my bad. I'll open a PR targetting the right branch.

**Comment by almahmoud** [comment]
Opened https://github.com/galaxyproject/galaxy/pull/9931 (didn't realize force pushing would close this PR)

**Pull request closed without merging**